### PR TITLE
#34044 -- Add the admin app filter on index pages in addition to the navbar

### PR DESCRIPTION
**Pull request by hoh**
Ticket: https://code.djangoproject.com/ticket/34044

This Pull Request contains two commits:
- The first refactors the admin quick search to make it independent from the navbar itself
- The second enables the quick search on index pages as well, in order to fix ticket 34044

## Screenshots

image
image
image
image
image
image
image

### CODE REVIEW

File: django/contrib/admin/static/admin/css/app_filter.css [hunk 1]
```diff
@@ -0,0 +1,53 @@
+
```
Comment (hoh) [inline django/contrib/admin/static/admin/css/app_filter.css:1]: Useless empty line

File: django/contrib/admin/static/admin/js/app_filter.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                }
+                // show/hide parent <TR>
+                o.node.parentNode.parentNode.style.display = displayValue;
+            }
+            if (!filterValue || matches) {
+                event.target.classList.remove('no-results');
+            } else {
+                event.target.classList.add('no-results');
+            }
+            sessionStorage.setItem('django.admin.navSidebarFilterValue', filterValue);
```
[the remark this answers is not in the document]
Comment (hoh) [reply django/contrib/admin/static/admin/js/app_filter.js:43]: The index page should not retain the search filter in the localStorage, this is only really useful in the navbar.

**Review by hoh — COMMENTED** [review]
Reviewed at the DjangoCon sprint with Thibault and Tom, looks good except for a few comments.

**Comment by knyghty** [comment]
Another option we discussed during the sprints is to simply add the sidebar to every page. I think this is actually the better solution after thinking about it some more.

The main issue is on the app pages, there's still no way to navigate to models for other apps but to go back to the index (or into a changelist view). It's quite annoying not having full navigation there in particular.

**Comment by felixxm** [comment]
@hoh Do you have time to keep working on this?

**Comment by hoh** [comment]
> @hoh Do you have time to keep working on this?

Yes, but only after mid-november.
I am OK with closing and re-opening if that makes managing issues/PRs easier, but the questions raised by @knyghty should not be forgotten.

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**